src/differentials/composite.jl

```diff
@@ -73,10 +73,16 @@ Base.getindex(comp::Composite, idx) = getindex(backing(comp), idx)
 # for Tuple
 Base.getproperty(comp::Composite, idx::Int) = unthunk(getproperty(backing(comp), idx))
 function Base.getproperty(
-    comp::Composite{P, <:NamedTuple{L}}, idx::Symbol
-) where {P, L}
-    # Need to check L directly, or else this does not constant-fold
-    idx ∈ L || return Zero()
+    comp::Composite{P, T}, idx::Symbol
+) where {P, L, T<:NamedTuple{L}}
+    # hasfield was added in v1.2
+    if VERSION ≥ v"1.2.0-DEV.272"
```

Comment:
Oh, good question. I'd never seen that before. I don't know. I could only find one place in base that uses it for `VERSION`.

Reply:
It definely used to be a thing pre-1.0
It seems to be less of a thing now, but i don't know if that is because somethign actually changed.

Running the experiment, it seem that the optimizer resolves it and removed the branch
```
julia> foo(x) = VERSION >=v"0.4" ? 1 : 2
foo (generic function with 1 method)

julia> @code_typed foo(1)
CodeInfo(
1 1 ─     return 1                                                                            │
  2 ─     $(Expr(:unreachable))::Union{}                                                      │
) => Int64
```